app/main.py

```diff
@@ -1,3 +1,51 @@
+from __future__ import annotations
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union[int, float, Distance]) -> Distance:
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        return Distance(self.km + other.km)
+
+    def __iadd__(self, other: Union[int, float, Distance]) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        if isinstance(other, (int, float)):
+            self.km += other
+        return self
+
+    def __mul__(self, other: Union[int, float]) -> Distance:
+        if isinstance(other, (int, float)):
+            return Distance(self.km * other)
+
+    def __truediv__(self, other: Union[int, float]) -> Distance:
+        if isinstance(other, (int, float)):
+            return Distance(round(self.km / other, 2))
+
+    def __lt__(self, other: Union[int, float, Distance]) -> bool:
+        return self.km < Distance(other).km
```

Comment:
what if other if int or float, these types have no attribute `.km`

Reply:
> what if other if int or float, these types have no attribute `.km`

I was trying to compare an instance with int, float and other instances and the result was generated correctly without any Error.  Please let me know what did I miss. 

def __lt__(self, other: Union[int, float, Distance]) -> bool:
    return self.km < other

distance = Distance(3)
distance1 = Distance(4)

print(distance.__lt__(distance1))
print(distance.__lt__(Distance(4)))
print(distance1.__lt__(4.05))
print(distance1.__lt__(2))

(True, <class '__main__.Distance'>)
(True, <class '__main__.Distance'>)
(True, <class 'float'>)
(False, <class 'int'>)